awsiotprotocol/loader_test.go

```diff
@@ -33,3 +35,13 @@ func TestByURL(t *testing.T) {
 		}
 	}
 }
+
+func TestByURLFailure(t *testing.T) {
+	_, err := ByURL("@@://@@@/@@.@@@")
+	assert.NotNil(t, err, "Should return error when a malformed url is given")
+	assert.Contains(t, err.Error(), "parse @@://@@@/@@.@@@: ")
```

Comment:
`assert.NotNil` seems to continue running the remaining test even if the assertion fails.
If `ByURL` returns nil, this line causes a segmentation fault.
(I guess this is one of the typical cases what https://golang.org/doc/faq#testing_framework is intended.)